pkg/kubernetes/snapshot/v1alpha1/kubernetes.go

```diff
@@ -0,0 +1,208 @@
+			return cli.VolumeSnapshots(k.namespace).Get(name, opts)
+			//return cli.StorageV1().StorageClasses().Get(name, opts)
+		}
+	}
+	if k.create == nil {
+		k.create = func(cli *clientset.OpenebsV1alpha1Client, snap *snapshot.VolumeSnapshot) (*snapshot.VolumeSnapshot, error) {
+			return cli.VolumeSnapshots(k.namespace).Create(snap)
+			//return cli.StorageV1().StorageClasses().Create(snap)
+		}
+	}
+	if k.del == nil {
+		k.del = func(cli *clientset.OpenebsV1alpha1Client, name string, opts *metav1.DeleteOptions) error {
+			return cli.VolumeSnapshots(k.namespace).Delete(name, opts)
+			//return cli.StorageV1().StorageClasses().Delete(name, opts)
+		}
+	}
+}
+
+// NewKubeClient returns a new instance of kubeclient meant for storageclass
+func NewKubeClient(opts ...KubeClientBuildOption) *Kubeclient {
+	k := &Kubeclient{}
+	for _, o := range opts {
+		o(k)
+	}
+	k.withDefaults()
+	return k
+}
+
+// WithClientSet sets the kubernetes client against
+// the kubeclient instance
+func WithClientSet(c *clientset.OpenebsV1alpha1Client) KubeClientBuildOption {
+	return func(k *Kubeclient) {
+		k.clientset = c
+	}
+}
+
+// WithKubeConfigPath sets the kubeConfig path
+// against client instance
+func WithKubeConfigPath(path string) KubeClientBuildOption {
+	return func(k *Kubeclient) {
+		k.kubeConfigPath = path
+	}
+}
+
+func (k *Kubeclient) getClientsetForPathOrDirect() (*clientset.OpenebsV1alpha1Client, error) {
+	if k.kubeConfigPath != "" {
+		return k.getClientsetForPath(k.kubeConfigPath)
+	}
+	return k.getClientset()
+}
+
+// getClientsetOrCached returns either a new
+// instance of kubernetes clientset or its
+// cached copy cached copy
+func (k *Kubeclient) getClientsetOrCached() (*clientset.OpenebsV1alpha1Client, error) {
+	if k.clientset != nil {
+		return k.clientset, nil
+	}
+
+	cs, err := k.getClientsetForPathOrDirect()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get clientset")
+	}
+	k.clientset = cs
+	return k.clientset, nil
+}
+
+// List returns a list of storageclass instances present in kubernetes cluster
+func (k *Kubeclient) List(opts metav1.ListOptions) (*snapshot.VolumeSnapshotList, error) {
+	cli, err := k.getClientsetOrCached()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to list storageclasses")
+	}
+	return k.list(cli, opts)
+}
+
+// Get return a storageclass instance present in kubernetes cluster
+func (k *Kubeclient) Get(name string, opts metav1.GetOptions) (*snapshot.VolumeSnapshot, error) {
+	cli, err := k.getClientsetOrCached()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to get storageclass {%s}", name)
+	}
+	return k.get(cli, name, opts)
+}
+
+// Create creates and returns a storageclass instance
+func (k *Kubeclient) Create(snap *snapshot.VolumeSnapshot) (*snapshot.VolumeSnapshot, error) {
+	if snap == nil {
+		return nil, errors.New("failed to create storageclass: nil storageclass object")
```

Comment:
nit: change `storageclass` to `snapshot` .

Reply:
Done